src/lib/ndpi_main.c

```diff
@@ -2102,6 +2102,10 @@ static void ndpi_init_protocol_defaults(struct ndpi_detection_module_struct *ndp
 			  "Thrift", NDPI_PROTOCOL_CATEGORY_RPC,
 			  ndpi_build_default_ports(ports_a, 0, 0, 0, 0, 0) /* TCP */,
 			  ndpi_build_default_ports(ports_b, 0, 0, 0, 0, 0) /* UDP */);
+  ndpi_set_proto_defaults(ndpi_str, 1 /* cleartext */, 0 /* nw proto */, NDPI_PROTOCOL_ACCEPTABLE, NDPI_PROTOCOL_SERVICE_LOCATION,
+              "Service Location Protocol", NDPI_PROTOCOL_CATEGORY_RPC,
+              ndpi_build_default_ports(ports_a, 427, 0, 0, 0, 0) /* TCP */,
```

Comment:
A default port for TCP, but the dissector is registered only on UDP traffic. Is that correct?

Reply:
Fixed. According to the RFC both L4 protocols can be used e.g. if the packet is too big to fit in a datagram, TCP is used instead.